### PR TITLE
ztp: Expose disk cleaning option through SiteConfig

### PR DESCRIPTION
**Pull request by irinamihai**
- add the automatedCleaningMode field to the SiteConfig CRD and by default set it to "disabled" (path to the new field is SiteConfig.Spec.Cluster.Node.automatedCleaningMode)
- the automatedCleaningMode field from a SiteConfig CR will be translated into the spec.automatedCleaningMode field in the generated BareMetalHost CR(s)
- update testcases to reflect this new field

### CODE REVIEW

File: ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site1-sno-du.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ spec:
   clusters:
   - clusterName: "site-sno-du-1"
     biosConfigRef:
-      filePath: "testSiteConfig/testHW.profile"
+      filePath: "../siteconfig-generator-kustomize-plugin/testSiteConfig/testHW.profile"
```
Comment (serngawy) [inline ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site1-sno-du.yaml:17]: why did you need to change that ?

File: ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site2-standard-du.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ spec:
   sshPrivateKeySecretRef:
     name: "sshPrvKey"
   biosConfigRef:
-    filePath: "testSiteConfig/testHW.profile"
+    filePath: "../siteconfig-generator-kustomize-plugin/testSiteConfig/testHW.profile"
```
Comment (serngawy) [inline ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site2-standard-du.yaml:16]: same

File: ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site1-sno-du.yaml (hunk 1 above)
Comment (irinamihai) [reply ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site1-sno-du.yaml:17]: I saw there was an error while executing 
```
siteconfig-generator]$  ./siteconfig-generator -manifestPath ../source-crs/extra-manifest ../siteconfig-generator-kustomize-plugin/testSiteConfig/site2-standard-du.yaml
2023/06/12 13:56:33 Error: could not build the entire SiteConfig defined by ../siteconfig-generator-kustomize-plugin/testSiteConfig/site2-standard-du.yaml: open testSiteConfig/testHW.profile: no such file or directory
```
because the testHW.profile didn't exist at the specified filePath and thought to address this.


I can remove this change, since it's outside the scope of this PR.
@tliu2021, what do you think?

Comment (tliu2021) [reply ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site1-sno-du.yaml:17]: The file path mentioned here is relative to the kustomization.yaml file, and this test file is intended to be utilized by the Makefile within the siteconfig-generator-kustomize-plugin. If you wish to use this test file outside of the siteconfig-generator-kustomize-plugin,  it may be beneficial to use the proposed path.

Comment (irinamihai) [reply ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site1-sno-du.yaml:17]: My reasoning was that since the [siteconfig-generator README](https://github.com/openshift-kni/cnf-features-deploy/tree/master/ztp/siteconfig-generator) suggests using the SiteConfig files from siteconfig-generator-kustomize-plugin/testSiteConfig, then all of them should successfully run as  siteconfig-generator parameters.

File: ztp/ran-crd/site-config-crd.yaml [hunk 3]
```diff
@@ -318,6 +318,14 @@ spec:
                                 - UEFI
                                 - UEFISecureBoot
                                 - legacy
+                            automatedCleaningMode:
+                              description: When set to disabled, automated cleaning will be avoided
+                                during provisioning and deprovisioning.
+                              type: string
+                              default: disabled
+                              enum:
+                                - metadata
```
Comment (serngawy) [inline ztp/ran-crd/site-config-crd.yaml:327]: what metadata ? I think this should be enable

Comment (irinamihai) [reply ztp/ran-crd/site-config-crd.yaml:327]: Hmm - I thought we are supposed to use the same definitions as in the original CRD, in this case BareMetaHost. This is how the BMH defines this field:
- https://github.com/metal3-io/baremetal-operator/blob/main/apis/metal3.io/v1alpha1/baremetalhost_types.go#L418
- https://github.com/metal3-io/baremetal-operator/blob/main/config/crd/bases/metal3.io_baremetalhosts.yaml#L77

Do we want to have the enum contain "enabled" and "disabled" instead of the original BMH definition? @imiller0 & @serngawy, kindly advise.

File: ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site1-sno-du.yaml (hunk 1 above)
Comment (tliu2021) [reply ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site1-sno-du.yaml]: I think your proposed change looks good.

Comment (irinamihai) [reply ztp/siteconfig-generator-kustomize-plugin/testSiteConfig/site1-sno-du.yaml]: Thanks, @tliu2021! We've decided to remove this change from this PR as it's unrelated. Might make a new PR with this small change in the future.

File: ztp/ran-crd/site-config-crd.yaml (hunk 3 above)
Comment (irinamihai) [reply ztp/ran-crd/site-config-crd.yaml:327]: Confirmed with baremetal folks that with the metadata option the partitioning table is removed, but the disk isn't fully wiped. I was wrong above and this is NOT related to this [field](https://github.com/metal3-io/baremetal-operator/blob/main/config/crd/bases/metal3.io_baremetalhosts.yaml#L245).

I updated the testcases accordingly. Thanks, @serngawy

**Review by serngawy — APPROVED** [review]
